distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSingleSignOn.java

```diff
@@ -0,0 +1,261 @@
+                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
+        clientConfig.update(clientSettings);
+        serverConfig.update(serverSettings);
+        assertThat(clientConfig.getProperties(), notNullValue());
+        assertThat(serverConfig.getProperties(), notNullValue());
+    }
+
+    private String getRedirectUrl(Response response) {
+        String fullUrl = "";
+
+        // We can either get a legit redirect from the header, or we can have javascript redirect us.
+        // Whenever javascript redirects us, we have to do ugly, fragile parsing of the HTML.
+        if (response.headers().hasHeaderWithName("Location")) {
+            fullUrl = response.header("Location");
+            // TODO: Figure out how to do this using xpath
+        } else if (response.getBody().asString()
+                .contains("<title>Redirect</title>")) { // Javascript redirect
+            Pattern pattern = Pattern.compile("encoded *= *\"(.*)\"", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(response.getBody().asString());
+            matcher.find();
+            fullUrl = matcher.group(1);
+        } else {
+            // TODO: this is unexpected. Log an error?
+        }
+        return fullUrl;
+    }
+
+    private String parseUrl(Response response) {
+        String url = getRedirectUrl(response);
+
+        // We want just the base url, not all of the parameters
+        if (url.contains("?")) {
+            url = url.substring(0, url.indexOf('?'));
+        }
+        return url;
+    }
+
+    private Map<String, String> parseParams(Response response) throws Exception {
+        String url = getRedirectUrl(response);
+
+        // RestAssured expects url query parameters to be in the form of a Dictionary.
+        List<NameValuePair> paramList = URLEncodedUtils.parse(new URI(url), "UTF-8");
+        HashMap<String, String> jsonParams = new HashMap<>();
+        for (NameValuePair param : paramList) {
+            jsonParams.put(param.getName(), param.getValue());
+        }
+        return jsonParams;
+    }
+
+    @Test
+    public void testBadUsernamePassword() throws Exception {
+
+        // Request search service
+        Response searchResponse = given().
+                redirects().follow(false).
+
+                expect().
+                statusCode(302).
+
+                when().
+                get(SEARCH_URL.getUrl());
+
+        // Pass bad credentials to IDP
+        Response IDPResponse = given().auth().preemptive().basic("definitely", "notright")
+                .param("AuthMethod", "up").params(parseParams(searchResponse))
+
+                // TODO: This should be a 401, not 500
+                .expect().statusCode(500).body(containsString("could not be authenticated"))
+
+                .when().get(parseUrl(searchResponse) + "/sso");
+    }
+
+    private void checkSaml(Map<String, String> samlParams) throws Exception {
```

Comment:
Do you have something in mind? validateSamlResponse maybe? That still seems a bit generic.